metabase/metabase
# Separate SMTP Connection Setup from Other Email Settings

**Problem:** Half of the email settings, the SMTP connection configuration, is **click-to-save**, while the other half, the additional email settings, is **auto-save**. This is a confusing paradigm for users.

**Context:**
- https://github.com/metabase/metabase/issues/32818

---
**Desired Behavior**/**Acceptance Criteria**

_Behaviors_
- [ ] SMTP connection configuration moved into separate setup placed behind a configuration card
- [ ] If SMTP is not configured, navigating to email settings auto-redirects to SMTP connection setup
  - [ ] After first time SMTP configuration, re-direct back to top-level email settings page to unveil further configuration options
- [ ] If SMTP is configured, show breadcrumbs above the SMTP connection configuration form
- [ ] All other functionality behaves as originally intended (saving, sending test emails, clearing the form, etc.)

_Tests_
- [ ] First time setup validated
  - [ ] On initial setup, auto-forward to SMTP connection configuration page from top-level email settings page (`/admin/settings/email` -> `/admin/settings/email/smtp`)
  - [ ] After successful first-time configuration, re-direct back to top-level page for additional settings (`/admin/settings/email/smtp` -> `/admin/settings/email`)
- SMTP configuration form
  - [ ] now placed behind a configuration card in top-level email settings (`/admin/settings/email`)
  - [ ] validated breadcrumbs show up on subsequent changes, but not the first time setup
  - [ ] no more re-directing when re-configuring an SMTP connection that is already setup
- [ ] All other behaviors work as intended (saving settings, sending test emails, clearing settings, etc.)

---
**How to Verify**

https://github.com/metabase/metabase/assets/22608765/e65e079c-7857-4b5b-8eab-caf3822e8f7f

1. Log in as admin
2. (if SMTP is already setup) Clear SMTP configuration
3. Go to top-level admin settings `/admin/settings/setup`
4. Navigate to `email` from the sidebar
5. See that you are auto-directed to SMTP configuration setup
6. Try invalid SMTP configuration and see that it shows an error
7. Setup a valid SMTP configuration and see that you are redirected to top-level email settings page where additional settings are revealed to you
8. See that the additional settings are auto-save

https://github.com/metabase/metabase/assets/22608765/fa367786-783f-48ea-8b5e-f3da4067dc4b

10. Navigate to top-level admin settings > `setup` in the sidebar
11. Navigate back to email in the sidebar
12. See that no re-direction occurs since it is already setup
13. See that SMTP configuration is now placed behind a configuration card
14. Click Edit Configuration and see that you are navigated to the SMTP connection form
